internal/httphandler/db_request_handler.go

```diff
@@ -70,6 +71,28 @@ func (d *dbRequestHandler) dbStatus(response http.ResponseWriter, request *http.
 	httputils.SendHTTPResponse(response, http.StatusOK, dbStatus)
 }
 
+func (d *dbRequestHandler) dbIndex(response http.ResponseWriter, request *http.Request) {
+	payload, respondedErr := extractVerifiedQueryPayload(response, request, constants.GetDBIndex, d.sigVerifier)
+	if respondedErr {
+		return
+	}
+	query := payload.(*types.GetDBIndexQuery)
+
+	dbIndex, err := d.db.GetDBIndex(query.DbName, query.UserId)
+	if err != nil {
+		httputils.SendHTTPResponse(
+			response,
+			http.StatusInternalServerError,
+			&types.HttpResponseErr{
+				ErrMsg: "error while processing '" + request.Method + " " + request.URL.String() + "' because " + err.Error(),
+			},
+		)
```

Comment:
Not all errors should return `http.StatusInternalServerError`. For example, permission, etc.
You have to switch between the known error types and select the right http status, and only for un-types errors return `http.StatusInternalServerError`. See e.g. [the config handler.](https://github.com/hyperledger-labs/orion-server/blob/38ec0896182ded380ae5d539a992e754385ff294/internal/httphandler/config_request_handler.go#L87)

Reply:
PS this also a problem in `func (d *dbRequestHandler) dbStatus()`, might as well fix it there too.